src/mysky/index.ts

```diff
@@ -132,10 +136,35 @@ export class MySky {
   // Constructors
   // ============
 
-  constructor(protected connector: Connector, permissions: Permission[], protected hostDomain: string) {
+  /**
+   * Creates a `MySky` instance.
+   *
+   * @param connector - The `Connector` object.
+   * @param permissions - The initial requested permissions.
+   * @param hostDomain - The domain of the host skapp.
+   * @param currentPortalUrl - The URL of the current portal. This is the portal that the skapp is running on, not the portal that may have been requested by the developer when creating a `SkynetClient`.
+   */
+  constructor(
+    protected connector: Connector,
+    permissions: Permission[],
+    protected hostDomain: string,
+    protected currentPortalUrl: string
+  ) {
+    if (MySky.instance) {
+      throw new Error("Trying to create a second MySky instance");
+    }
+
     this.pendingPermissions = permissions;
   }
```

Comment:
I honestly don't have the bandwidth to be thinking about non-urgent things right now. This was not introduced in this PR, and in my opinion is "low in urgency, low in importance". Better to get to it later when we can have a proper think about it (and maybe Karol or Michał can provide input). If this is important to you right now and you have time then feel free to contribute a commit.

I'll save your comment in the associated issue. Thanks for writing it out - I just can't get pulled into random things right now.

> I'd prefer to solve this here rather than postponing it to a F/U which might fall through the cracks.

I am aware of the outstanding issues in the projects I am responsible for. It's not really a matter of falling through the cracks, but of being chronically overworked for at least the past year.